src/components/fuel/index.tsx

```diff
@@ -89,101 +100,142 @@ export default function FuelForm() {
+                    min={10}
+                    placeholder={t("input.amount_paid")}
+                    value={data["Amount Paid"]}
+                    onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                      setData({ ...data, "Amount Paid": e.target.value })
+                    }
+                    required={true}
+                  />
+                  <p className="text-black absolute z-10 bottom-[0.5rem]  right-[1.2rem]">
+                    €
+                  </p>
+                </FormWrapper>
+                <FormWrapper>
+                  <CommonLabel input="text">
+                    {t("input.captain_list")}
+                  </CommonLabel>
+                  <SelectCaptain
+                    disabled={!!searchParams.get("captainId")}
+                    value={data["Captain"]}
+                    onChange={(captainId) =>
+                      setData({ ...data, Captain: captainId })
+                    }
+                  />
+                </FormWrapper>
+              </div>
+              <div className="flex md:flex-row flex-col justify-between w-full md:mt-6 mt-0">
+                <FormWrapper>
+                  <CommonLabel input="text">{t("input.port")}</CommonLabel>
+                  <CommonSelect
+                    id="port"
+                    name="port"
+                    data={PORTS}
+                    value={data["Port"]}
+                    onChange={(e) => setData({ ...data, Port: e.target.value })}
+                    required
+                  />
+                </FormWrapper>
+                <FormWrapper>
+                  <CommonLabel input="text">{t("input.pin")}</CommonLabel>
+                  <div className="relative w-full">
+                    <div className="absolute inset-y-0 right-0 flex items-center px-2">
+                      {showPin ?
+                        <div onClick={() => setShowPin(!showPin)} className="cursor-pointer">
+                          <svg className="h-6 text-gray-700" fill="none" xmlns="http://www.w3.org/2000/svg"
+                            viewBox="0 0 640 512">
+                            <path fill="currentColor"
+                              d="M320 400c-75.85 0-137.25-58.71-142.9-133.11L72.2 185.82c-13.79 17.3-26.48 35.59-36.72 55.59a32.35 32.35 0 0 0 0 29.19C89.71 376.41 197.07 448 320 448c26.91 0 52.87-4 77.89-10.46L346 397.39a144.13 144.13 0 0 1-26 2.61zm313.82 58.1l-110.55-85.44a331.25 331.25 0 0 0 81.25-102.07 32.35 32.35 0 0 0 0-29.19C550.29 135.59 442.93 64 320 64a308.15 308.15 0 0 0-147.32 37.7L45.46 3.37A16 16 0 0 0 23 6.18L3.37 31.45A16 16 0 0 0 6.18 53.9l588.36 454.73a16 16 0 0 0 22.46-2.81l19.64-25.27a16 16 0 0 0-2.82-22.45zm-183.72-142l-39.3-30.38A94.75 94.75 0 0 0 416 256a94.76 94.76 0 0 0-121.31-92.21A47.65 47.65 0 0 1 304 192a46.64 46.64 0 0 1-1.54 10l-73.61-56.89A142.31 142.31 0 0 1 320 112a143.92 143.92 0 0 1 144 144c0 21.63-5.29 41.79-13.9 60.11z">
+                            </path>
```

Comment:
SVG needs to be its own component.

And then this logic need to be in the SecretInput component